sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/DetectLanguageClient.java

```diff
@@ -0,0 +1,126 @@
+            text, countryHint));
+        return detectBatchLanguagesWithResponse(languageInputs, null, context).map(response ->
+            new SimpleResponse<>(response, response.getValue().iterator().next()));
+    }
+
+    Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectLanguagesWithResponse(List<String> textInputs,
+        String countryHint, Context context) {
+        Objects.requireNonNull(textInputs, "'textInputs' cannot be null.");
+        List<DetectLanguageInput> detectLanguageInputs = mapByIndex(textInputs, (index, value) ->
+            new DetectLanguageInput(index, value, countryHint));
+
+        return detectBatchLanguagesWithResponse(detectLanguageInputs, null, context);
+    }
+
+    Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectBatchLanguagesWithResponse(
+        List<DetectLanguageInput> textInputs, TextAnalyticsRequestOptions options, Context context) {
+        Objects.requireNonNull(textInputs, "'textInputs' cannot be null.");
+
+        final LanguageBatchInput languageBatchInput = new LanguageBatchInput()
+            .setDocuments(textInputs.stream().map(detectLanguageInput -> new LanguageInput()
+                .setId(detectLanguageInput.getId()).setText(detectLanguageInput.getText())
+                .setCountryHint(detectLanguageInput.getCountryHint())).collect(Collectors.toList()));
+
+        return service.languagesWithRestResponseAsync(
+            languageBatchInput, options == null ? null : options.getModelVersion(),
+            options == null ? null : options.showStatistics(), context)
+            .doOnSubscribe(ignoredValue -> logger.info("A batch of language input - {}", textInputs.toString()))
+            .doOnSuccess(response -> logger.info("A batch of detected language output - {}", response.getValue()))
+            .doOnError(error -> logger.warning("Failed to detected languages - {}", error))
+            .map(response -> new SimpleResponse<>(response, toDocumentResultCollection(response.getValue())));
+    }
+
+    /**
+     * Helper method to convert the service response of {@link LanguageResult} to {@link DocumentResultCollection}.
+     *
+     * @param languageResult the {@link LanguageResult} returned by the service.
+     *
+     * @return the {@link DocumentResultCollection} of {@link DetectLanguageResult} to be returned by the SDK.
+     */
+    private DocumentResultCollection<DetectLanguageResult> toDocumentResultCollection(
+        final LanguageResult languageResult) {
+
+        final List<DetectLanguageResult> detectLanguageResults = new ArrayList<>();
+        for (DocumentLanguage documentLanguage : languageResult.getDocuments()) {
+            DetectedLanguage primaryLanguage = null;
+            if (documentLanguage.getDetectedLanguages().size() >= 1) {
+                com.azure.ai.textanalytics.implementation.models.DetectedLanguage detectedLanguageResult =
```

Comment:
It's confusing to have the same name for implementation and public model classes.

Reply:
Models in the implementation folder are auto-generated models. Some of the public model classes adopt the name but they are different. We want to explore the same name as `DetectedLanguage` and also keep no-change on the auto-generated code. That is the only way we found. Love to see another option.